index.d.ts

```diff
@@ -5,9 +5,9 @@ export = OpenApiEnforcerMiddleware
 declare class OpenApiEnforcerMiddleware {
     constructor (definition: string|object, options?:OpenApiEnforcerMiddleware.Options );
 
-    controllers (controllersDirectoryPath: string|object, ...dependencyInjection: any): Promise<object>;
-    middleware (): RequestHandler;
-    mocks (controllersDirectoryPath: string|object|undefined, automatic?: boolean, ...dependencyInjection: any): Promise<object>;
+    controllers<T extends unknown[]> (controllersDirectoryPath: string | OpenApiEnforcerMiddleware.Controllers, ...dependencyInjection: T): Promise<object>;
```

Comment:
Please educate me on this one. Why `<T extends unknown[]>` instead of using `any[]`?

Reply:
The short answer is that `any` isn't type safe so I always default to using `unknown` instead of `any`. It's kind of like using `const` by default even though you could use `let` to declare all variables in javascript.

You can read more about unknown vs any on [this stackoverflow answer](https://stackoverflow.com/a/51439876/7542561) or on the [Typescript 3.0 RC](https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type).

Reply:
There's not much of a difference between the two in this particular situation but utilizing the spread operator and `unknown` type is more type safe than using `any[]`.